### ISSUE TITLE
Crash after harvesting resin

### ISSUE THREAD

**Issue by Featheredbat** [anegostudios/VintageStory-Issues#4789]
### Game Version

1.20.0-rc.6

### Platform

Windows

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

After right-clicking to harvest resin, the resin did not appear in my inventory despite vanishing off the tree, and a couple of seconds later the game crashed

### How to reproduce

Possibly related to harvesting the resin, but not sure

### Logs
```
Running on 64 bit Windows 10.0.19045.0 with 32708 MB RAM
Game Version: v1.20.0-rc.6 (Unstable)
12/25/2024 1:59:35 AM: Critical error occurred
Loaded Mods: game@1.20.0-rc.6, creative@1.20.0-rc.6, survival@1.20.0-rc.6
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.ClientChunkData.EmptyAndReuseArrays(List`1 datas) in VintagestoryLib\Client\Model\ClientChunkData.cs:line 197
   at Vintagestory.Common.ChunkDataPool.FreeArraysAndReset(ChunkData cdata) in VintagestoryLib\Common\Datastructures\ChunkDataPool.cs:line 89
   at Vintagestory.Common.WorldChunk.Dispose() in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 461
   at Vintagestory.Client.NoObf.SystemUnloadChunks.HandleChunkUnload(Packet_Server packet) in VintagestoryLib\Client\Systems\UnloadChunks.cs:line 87
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 786
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 161
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```

**Comment by Craluminum2413** [comment]
Duplicate of https://github.com/anegostudios/VintageStory-Issues/issues/4692

**Comment by radfast** [comment]
Thanks @Featheredbat for reporting.  I apologise you had a crash, I thought we'd fixed this one properly by now, but apparently not...

It is not specifically related to gathering resin, it is something that happens very rarely (like 1 game in 1000, something like that) due to a bug deep inside the engine, essentially two systems in the engine are trying to unload and compress a chunk far from player, at approximately the same time.  It relates to chunks far away, not the chunk the player is currently in.

You can safely continue to play 1.20.0-rc.6 as normal, chances of this happening again to you specifically are very low

**Comment by Featheredbat** [comment]
Okay great thank you!!

**Comment by radfast** [comment]
The issue is properly fixed in upcoming 1.20.0-rc.7.   Thanks again for reporting (despite the duplicate flag), it was very helpful to me to see an example of it still occurring in 1.20.0-rc.6.
